src/models/order.ts

```diff
@@ -0,0 +1,36 @@
+import mongoose, { Document, Schema } from 'mongoose';
+
+const orderItemSchema = new Schema({
+  name: { type: String, required: true },
+  quantity: { type: Number, required: true },
+  price: { type: Number, required: true },
+});
+
+const orderSchema = new Schema({
+  user: { type: Schema.Types.ObjectId, ref: 'User', required: true },
+  cookId: { type: Schema.Types.ObjectId, ref: 'Cook' },
+  orderItems: [orderItemSchema],
+  deliveryFee: { type: Number, required: true },
+  totalOrderPrice: { type: Number, required: true },
+  paymentMethodType: { type: String, required: true },
+  isPaid: { type: Boolean, default: false },
+  paidAt: { type: Date },
+});
+
+
+interface IOrder extends Document {
+  user: string;
+  cookId?: string;
+  orderItems: {
+    name: string;
+    quantity: number;
+    price: number;
+  }[];
+  deliveryFee: number;
+  totalOrderPrice: number;
+  paymentMethodType: string;
+  isPaid: boolean;
+  paidAt?: Date;
+}
+
+export const Order = mongoose.model<IOrder>('Order', orderSchema);
```

Comment:
**The updated code:** 

`
import { Schema, model, Document, Types } from 'mongoose';

export interface IOrder extends Document {
  user: Types.ObjectId;
  cookId: string;
  orderItems: Array<{ product: Types.ObjectId; quantity: number }>;
  deliveryFee: number;
  totalOrderPrice: number;
  paymentMethodType: string;
}

const orderSchema = new Schema<IOrder>(
  {
    user: { type: Schema.Types.ObjectId, ref: 'User', required: true },
    cookId: String,
    orderItems: [
      {
        product: {
          type: Types.ObjectId,
          ref: 'Dish',
        },
        quantity: {
          type: Number,
          default: 1,
        },
      },
    ],
    deliveryFee: { type: Number, default: 0 },
    totalOrderPrice: { type: Number, default: 0 },
    paymentMethodType: { type: String, default: 'Cash' },
  },
  { timestamps: true }
);

const Order = model<IOrder>('Order', orderSchema);
export default Order;

`

![image](https://github.com/AsliSema/byte-and-bite/assets/86200305/e1f8b819-dff7-402a-b457-ff25d8efae64)